src/Eloquent/Mixins/BuilderReturning.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tpetry\PostgresqlEnhanced\Eloquent\Mixins;
+
+use Closure;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Model;
+use Tpetry\PostgresqlEnhanced\Support\Helpers\ModelHelper;
+
+/** @mixin \Illuminate\Database\Eloquent\Builder */
+class BuilderReturning
+{
+    public function deleteReturning(): Closure
+    {
+        return function (array $returning = ['*']): Collection {
+            /** @var \Illuminate\Database\Eloquent\Builder $this */
+            $time = $this->getModel()->freshTimestampString();
+            $modelHelper = app(ModelHelper::class);
+
+            if ($deletedAtColumn = $modelHelper->getDeletedAtColumn($this->getModel())) {
+                return $this->updateReturning([$deletedAtColumn => $time], $returning);
+            }
+
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->deleteReturning(null, $returning)->all()
+            )->each(fn (Model $model) => $model->exists = false);
+        };
+    }
+
+    public function insertOrIgnoreReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->insertOrIgnoreReturning($values, $returning)->all()
+            );
+        };
+    }
+
+    public function insertReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->insertReturning($values, $returning)->all()
+            );
+        };
+    }
+
+    public function insertUsingReturning(): Closure
+    {
+        return function (array $columns, $query, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->insertUsingReturning($columns, $query, $returning)->all()
+            );
+        };
+    }
+
+    public function updateFromReturning(): Closure
+    {
+        return function (array $values, array $returning = ['*']): Collection {
+            /* @var \Illuminate\Database\Eloquent\Builder $this */
+            return $this->hydrate(
+                $this->applyScopes()->getQuery()->updateFromReturning($values, $returning)->all()
```

Comment:
as we are at EloquenBuilder level, I think that it would make sense to also update the timestamp here : 
```suggestion
                $this->applyScopes()->getQuery()->updateFromReturning($this->addUpdatedAtColumn($values), $returning)->all()
```

Reply:
Sounds fair but Laravel does not actually provides an `updateFrom` at Eloquent level but only at Database level, which explains why timestamps are not updated.